src/workers/sequencingDiagnostics.ts

```diff
@@ -162,3 +167,183 @@ function generateAbsoluteTimeStringDiagnostics(
+    console.log({ discoveredArgs, expectedArgMap, foundDiagnostics: diagnostics });
+    return diagnostics;
+  } else if (tsc.isStringLiteral(argument)) {
+    if (commandName === 'GROUND_EVENT') {
+      return [];
+    } else {
+      console.error('Found a command with an argument of an uncaught type', commandNode.getText(), argument.getText());
+      return [
+        makeDiagnostic(
+          'Uncaught argument object property type',
+          CustomCodes.UncaughtArgumentType,
+          sourceFile,
+          argument,
+        ),
+      ];
+    }
+  } else {
+    console.error('Found a command with an argument of an uncaught type', commandNode.getText(), argument.getText());
+    return [makeDiagnostic('Uncaught argument type', CustomCodes.UncaughtArgumentType, sourceFile, argument)];
+  }
+}
+
+/**
+ * Generate diagnostics
+ *
+ * Extracts the commands by starting with the timing letters and recursing up the parent tree
+ *
+ * Let -> represent "access parent"
+ *
+ * A, E, R all work like this:
+ *  A -> A("time") -> A("time").COMMAND -> A("time").COMMAND({args})
+ *
+ * C works like this:
+ *  C -> C.COMMAND -> C.COMMAND({args})
+ *
+ * Any node (.*)\.COMMAND has the children:
+ *  [$1.kind ($1), DotToken (.), Identifier (COMMAND)]
+ * And we can match on that!!
+ * The first setup that looks like this up the parent tree will always be the command.
+ *
+ * Furthermore, the parent of this node will have the SyntaxList where the text will be the string of an object.
+ *  We should parse this into an object and validate that every key:
+ *  1. Is supposed to be there
+ *  2. Is within the correct range
+ *
+ * @param sourceFile The starting point for analysis
+ */
+function findAndValidateArguments(sourceFile: tsc.SourceFile, commandDict: CommandDictionary): ResponseDiagnostic[] {
+  // Finds a node of the format `(.*)\.COMMAND`
+  // It should be a PropertyAccessExpression
+  const commandNodes = findCommandNodes(sourceFile).filter(node => tsc.isPropertyAccessExpression(node));
+
+  let diagnostics = [];
+
+  for (const commandNode of commandNodes) {
+    // The command name will be the identifier child
+    const commandName = commandNode.getChildren().reverse().find(tsc.isIdentifier);
+
+    if (commandName === undefined) {
+      console.error('Found a bad command node: ', commandNode.getText());
+      continue;
+    }
+
+    const commandDictEntry = commandDict.fswCommandMap[commandName.getText()];
+
+    if (commandDictEntry === undefined) {
+      console.error('Failed to find a command dict entry for command', commandName.getText());
+    }
+
+    // The parent of the commandNode will be a CallExpression (in the case of `C.COMMAND_NAME(...)`)
+    //  Or it won't (in the no argument + no call case)
+    // Note that it doesn't matter if arguments or properites are missing or not - tsc will handle that for us.
+    if (tsc.isCallExpression(commandNode.parent)) {
```

Comment:
if you have this:

`C.REQUEST('string', C.PREHEAT_OVEN({temperature : 10}))`

it throws off the code below. You get

`command_arguments : ['string, C.PREHEAT_OVEN({temperature : 10})] `

instead of 

`command_arguments : [temperature :10]`

Reply:
Ah yep, I wasn't aware that was a possibility! This could be integrated as another check in the filter function or on the list of command candidates before we operate on them.

I believe it could be done by asserting that any detected command does not have any other command as a child node. In the case of 

`C.REQUEST('string', C.PREHEAT_OVEN({temperature : 10}))`

We can see that the two call expressions are children:

https://ts-ast-viewer.com/#code/MIOgSgogigqhDKAVAFAcgM4BcBOBLAdgOaoA0ABKAAqQASEAgogPoDyAahAHLIDemApgFsADv2wBDTAFds-MgC4yARgAMAXwCUGoA

Reply:
However, this is definitely a one case patch, and what this really needs is robust testing to ensure it works on everything in the possibility space.